### PR TITLE
Rest tipps crossref

### PR DESCRIPTION
**Pull request by hornmo**
- Add PUT/PATCH REST endpoints for Sources
- Add POST endpoint for ReviewRequests
- Move `IntegrationController.processHistoryEvents` to TitleHistoryService
- Move `IntegrationController.addPublisherHistory` to TitleLookupService
- Catch `MultipleComponentsMatchedException` in IntegrationController actions
- Add UpdateToken generation for `PackageController.update()`
- Catch validation errors for title generation in `PackageController.save()`
- Extend TIPP coverageDepth validation/upsert
- Add PUT endpoint for package TIPPs (similar to `IntegrationController.crossReferencePackage`)
- Fix typo in ReviewsController
- Add allocatedGroups to ReviewRequest default embeds
- Add ReviewRequest allocatedGroups handling for ComponentLookupService
- Add type filter in GlobalController
- Fix CuratoryGroup name validation
- Add flexible handling for RefdataValues in `RestMappingServive.updateObject` & `ClassUtils.setRefdataIfPresent`

### CODE REVIEW

File: server/gokbg3/grails-app/init/gokbg3/BootStrap.groovy [hunk 1]
```diff
@@ -203,6 +203,13 @@ class BootStrap {
       }
       log.debug("${ctr} components updated");
 
+    log.info("GoKB remove usused refdata");
+      def rr_std = RefdataCategory.lookup('ReviewRequest.StdDesc', 'RR Standard Desc 1')
+
+      if (rr_std) {
+        rr_std.delete()
```
Comment (VolkerHBZ) [inline server/gokbg3/grails-app/init/gokbg3/BootStrap.groovy:210]: Cleaning known junk at startup seems weird. Where does this RefdataCategory come from in the first place? Why it's written to the database at all and why isn't that avoided?

File: server/gokbg3/src/main/groovy/com/k_int/ClassUtils.groovy [hunk 2]
```diff
@@ -152,7 +152,7 @@ class ClassUtils {
         v = RefdataCategory.lookup(cat, value)
       }
     }
-    else if (value && cat) {
+    else if (value instanceof Integer && cat) {
       try {
         def candidate = RefdataValue.get(value)
```
Comment (VolkerHBZ) [inline server/gokbg3/src/main/groovy/com/k_int/ClassUtils.groovy:157]: Not sure, but afaik get() works on Database ids of type Long, not Integer.

File: server/gokbg3/grails-app/controllers/gokbg3/UrlMappings.groovy [hunk 3]
```diff
@@ -16,6 +16,7 @@ class UrlMappings {
     group "/rest", {
       post "/packages/$id/retire"(controller: 'package', namespace: 'rest', action:'retire')
       post "/packages/$id/tipps"(controller: 'package', namespace: 'rest', action: 'addTipps')
+      put "/packages/$id/tipps"(controller: 'package', namespace: 'rest', action: 'updateTipps')
       "/packages/$id/$action"(controller: 'package', namespace: 'rest')
```
Comment (VolkerHBZ) [inline server/gokbg3/grails-app/controllers/gokbg3/UrlMappings.groovy:21]: just add `patch "/packages/$id/tipps"(controller: 'package', namespace: 'rest', action: 'updateTipps')` as well

File: server/gokbg3/grails-app/init/gokbg3/BootStrap.groovy (hunk 1 above)
Comment (hornmo) [reply server/gokbg3/grails-app/init/gokbg3/BootStrap.groovy:210]: The value was added via BootStrap.groovy before, which was only removed in a previous patch. We can therefore assume it was at some point created in all existing instances so has to be removed everywhere.

File: server/gokbg3/src/main/groovy/com/k_int/ClassUtils.groovy (hunk 2 above)
Comment (hornmo) [reply server/gokbg3/src/main/groovy/com/k_int/ClassUtils.groovy:157]: It does work with both actually. GORM defines the type as `java.io.Serializable` internally.

**Review by VolkerHBZ — APPROVED** [review]

File: server/gokbg3/grails-app/controllers/gokbg3/UrlMappings.groovy (hunk 3 above)
Comment (hornmo) [reply server/gokbg3/grails-app/controllers/gokbg3/UrlMappings.groovy:21]: Agreed. PATCH now has the behaviour of `addOnly` by default, where missing TIPPs are simply ignored.

**Pull request merged**